### PR TITLE
[GH-23419] Migrate "components/admin_console/set_by_env.jsx" to Typescript

### PR DESCRIPTION
**Pull request by tejaskarelia17**
#### Summary
In this PR, completes migration of `components/admin_console/set_by_env.jsx` to TypeScript.

#### Ticket Link
Fixes [GitHub issue #23419](https://github.com/mattermost/mattermost-server/issues/23419)
https://mattermost.atlassian.net/browse/MM-52841

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/admin_console/set_by_env.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
 
-export default function SetByEnv() {
+const SetByEnv: React.FunctionComponent = (): JSX.Element => {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/set_by_env.tsx:7]: we actually don't have to convert this to an anonymous function. its fine as it was before with default export. This conversion might actually be changing the file extension :D

**Review by M-ZubairAhmed — DISMISSED** [review]

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
Please check the above comment

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by sinansonmez — APPROVED** [review]
LGTM 👍

**Review by yasserfaraazkhan — APPROVED** [review]
thank you for the changes 
e2e have passed 

https://github.com/mattermost/mattermost-server/pull/23515

**Pull request merged**